src/pages/api/vesting/[address].ts

```diff
@@ -1,19 +1,85 @@
 import type { NextApiRequest, NextApiResponse } from 'next';
 import DATA from './data.json';
-import { addressConvert } from '@common/utils/addresses';
+import LOCKUP from './lockups.json';
+import {
+  addressConvert,
+  fetchCurrentStacksBlock,
+  fetchLegacyExplorerVestingData,
+} from '@common/utils/addresses';
+import BN from 'bn.js';
 
 const addresses = new Set(DATA);
 
+interface LockupData {
+  [key: string]: {
+    [key: string]: number;
+  };
+}
+
+async function fetchActualBalance(address: string) {
+  try {
+    const res = await fetch(`https://stacks-node-api.stacks.co/v2/accounts/${address}?proof=0`);
+    const data = await res.json();
+    return data;
+  } catch (e) {
+    throw Error(`Can't fetch actual balance: ${e.message}`);
+  }
+}
+
+async function fetchAverageBlockTime() {
+  const res = await fetch('https://blockchain.info/q/interval');
+  const data = await res.text();
+  return data;
+}
+
+export type VestingAddressData =
+  | {
+      found: true;
+      data: { [key: string]: number };
+      balance: string;
+      currentHeight: number;
+      lockedBalance: number;
+      averageBlockTime: string;
+      originalTotal: number | null;
+    }
+  | { found: false }
+  | { error: string };
+
 export default async function vestingAddressHandler(
   { query: { address } }: NextApiRequest,
-  res: NextApiResponse<{ found: boolean } | { error: string }>
+  res: NextApiResponse<VestingAddressData>
 ) {
   try {
     if (address && typeof address === 'string') {
       const result = addressConvert(address);
+
       const found = addresses.has(result.mainnet.BTC || result.mainnet.STACKS);
+      const data = (LOCKUP as LockupData)[result.mainnet.BTC];
+
       if (found) {
-        res.status(200).json({ found: true });
+        const [coreNodeData, currentHeight, legacyData, averageBlockTime] = await Promise.all([
+          fetchActualBalance(result.mainnet.STACKS),
+          fetchCurrentStacksBlock(),
+          fetchLegacyExplorerVestingData(result.mainnet.STACKS),
+          fetchAverageBlockTime(),
+        ]);
+
+        let lockedBalance = 0;
+        Object.keys(data).forEach((height, index) => {
+          if (parseInt(height) > currentHeight) {
+            lockedBalance += Object.values(data)[index];
+          }
+        });
```

Comment:
```suggestion
        const lockedBalance = Object.keys(data).reduce((acc, height) => {
          if (parseInt(height) > currentHeight) return acc + parseInt(height);
          return parseInt(height);
        }, 0);
```
can do more functionally something like ^ (untested)

Reply:
Thank you :)